### DISCUSSION TITLE
Problem with the upper-ci for a 30 years datasets

### DISCUSSION THREAD

**Discussion by LilRamboll** [georgebv/pyextremes#56] [Q&A]
Hi,

I am encouting an problem with the upper-ci calculated with pyextreme.
I am calculating the return period of precipitation of a 30 year daily dataset for the period of 1991-2020, here is the following code:

```
TP_EMO = np.loadtxt("Gardanne_pr_1991-2020_EMO.csv")
dates_EMO = [datetime(1991,1,1)+n*timedelta(hours=24) for n in range(10958)]

RR_UERRA = pd.Series(TP_EMO,index=dates_EMO)

RR_UERRA = (
    RR_UERRA
    .sort_index(ascending=True)
    .astype(float)
    .dropna()
)

model=EVA(data=RR_UERRA)
print(model)
model.get_extremes(
    method="BM",
    extremes_type="high",
    block_size="365.2425D",
    errors="raise",
    )
print(model)
model.plot_extremes()
model.fit_model()
print(model)

model.plot_diagnostic(alpha=0.95)

model.plot_return_values(
    return_period=np.logspace(0.01, 2, 100),
    return_period_size="365.2425D",
    alpha=0.95,
)

summary = model.get_summary(
    return_period=[2, 5, 10, 25, 50, 100],
    alpha=0.95,
    n_samples=1000,
)
print(summary)
```

And here is the result :
       
```
 return period   return value    lower ci      upper ci
 2                 51.997317   40.328411  6.366540e+01
 5                 76.366393   62.001047  4.059330e+04
 10                96.723240   74.228410  8.974462e+06
 25               128.522617   89.655924  6.869750e+09
 50               157.462879  100.629528  1.030623e+12
 100              191.659978  112.882209  1.640525e+14
```

As you can see, the return value and lower-ci are correct. However, the upper-ci is wrong.
I tried using different dataset but this issues occurs with all 30 years dataset.
There seems to be an issues with the distribution of the upper-ci, did anyone else encounter this issues and is there a solution ?

Thanks,
Lilian

**Comment by georgebv** [comment]
This is a late response, but typically what you see is indicative of a poor fit - meaning, your data does not follow the GEV distribution. Your return values are naturally bounded on the lower side, but the upper bound screams "poor fit". You should post PDF and CDF charts (use the diagnostic plot in pyextremes).

Keep in mind that GEVD models distribution of block maxima for a **random** (this is the key) signal. Things like a trend, periodic carrier signal, censorship can cause this.

Also, for 30 years of data I would not rely on any projection beyond ~50 years.
